### PR TITLE
[benchmark] Add mock floating-point types for conversion benchmarks

### PR DESCRIPTION
**Pull request by xwu**
This is a companion PR which includes, for benchmarking purposes, a protocol `MockBinaryFloatingPoint` with default implementations that allows easy creation of "newtypes" that wrap existing binary floating-point types.

This permits performance testing of generic conversions between binary floating-point types where one or both are not known to the standard library, a scenario which is encountered in real-world libraries such as SwiftUI that convert between `CGFloat` and standard library types.

### CODE REVIEW

**Comment by xwu** [comment]
@swift-ci benchmark

File: benchmark/single-source/FloatingPointConversion.swift [hunk 1]
```diff
@@ -21,8 +21,120 @@ public let FloatingPointConversion = [
...
+  func isLessThanOrEqualTo(_ other: Self) -> Bool {
+    _value.isLessThanOrEqualTo(other._value)
+  }
+  mutating func round(_ rule: FloatingPointRoundingRule) { _value.round(rule) }
+}
+
+struct MockFloat32: MockBinaryFloatingPoint {
+  var _value: Float
+  init(_ _value: Float) { self._value = _value }
+}
```
Comment (benrimmington) [inline benchmark/single-source/FloatingPointConversion.swift:131]: (`MockFloat32` is currently unused.)

**Review by benrimmington — COMMENTED** [review]
I'm not the best person to review this benchmark.

Can the warnings about naming (>40 chars) and workload (>1000 μs) be ignored?

**Comment by xwu** [comment]
@benrimmington Thanks for taking a look. I'll see if someone else can chime in as well on this.

The warnings about the name length look to be false positives, as the suggested remedy is to use dots to break up the name--which we already do! The workload issue is definitely something I'm paying attention to; in all cases, the benchmarks are intended to perform under 1000 µs imminently, but if I change the workload upfront, then some of them will be very noisy as they'll run in the <20 µs range after the improvements land.

**Comment by xwu** [comment]
Since I'm eager to see how much these benchmarks can be improved on CI with the mentioned PR, I'm going to go ahead and merge this. Happy to work with folks in the know to correct any naming issues that come up.

**Comment by xwu** [comment]
@swift-ci smoke test and merge

**Comment by xwu** [comment]
@swift-ci smoke test and merge

**Pull request merged**